napari/components/dims.py

```diff
@@ -225,6 +231,53 @@ def set_range(
                     ax = assert_axis_in_bounds(int(ax), self.ndim)
                     full_range[ax] = r
                 self.range = full_range
+        if restore_point is not None:
+            new_point = self._get_point_in_ndim(restore_point)
+            self.set_point(range(self.ndim), new_point)
+
+    @staticmethod
+    def rounded_division(min_val, max_val, precision):
+        return int(((min_val + max_val) / 2) / precision) * precision
+
+    def _get_point_in_ndim(
+        self,
+        point: Union[Union[int, float], Sequence[Union[int, float]]],
+    ):
+        """Get the point within the current dimensions of dimensions.
+        Coordinates outside the current dim are removed.
+        If new dimensions need to be added the midpoint of the respective range
+        is chosen for the dimensions point value.
+
+        Parameters
+        ----------
+        point : tuple
+            Point check if it is within the current range.
+
+        Returns
+        ---------
+        point: tuple
+            Tuple within the current dimension
+        """
+        # This logic block defines the behavior when layers are added or
+        # removed. Its goal is it to keep the currently selected view
+        # selected by the user.
```

Comment:
Consider removing this comment as it doesn't make much sense within the `Dims` class.

Reply:
Removed it